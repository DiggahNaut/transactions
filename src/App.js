import React, {Component} from 'react';
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import 'react-tabs/style/react-tabs.css';
import TransactionsTable from './components/TransactionsTable';
import ProjectList from './components/ProjectList';
import PaymentMethodList from './components/PaymentMethodList';
import PaymentMethodChart from './components/PaymentMethodChart';
import transactionsData from './transactions.json';

// Функция возвращает массив платёжных систем с подсчётом всех совершённых транзакций по каждой системе и сортировкой по убыванию.
function getPaymentData(props) {
  const paymentMethods = [];

  for (var i = 0; props.length > i; i++) {
    var name = props[i].transaction.payment_method.name;
    var key = props[i].transaction.payment_method.id;
    var amountOfUse = 1;

    if (paymentMethods[key]) {
      amountOfUse = paymentMethods[key].amountOfUse + 1;
    }

    paymentMethods[key] = {
      amountOfUse: Number(amountOfUse),
      label: name
    };
  }

  return paymentMethods.sort((a, b) => b.amountOfUse - a.amountOfUse);
}

class App extends Component {
  render() {
    // paymentData нужны для построения рейтинга и графика платёжных систем.
    // Решил формировать массив данных для компонентов PaymentMethodList/PaymentMethodChart здесь.
    // Чтобы не делать этого каждый раз в этих компонентах.
    const paymentData = getPaymentData(transactionsData);

    return (
      <div className="container">
        <Tabs>
          <TabList>
            <Tab>Транзакции</Tab>
            <Tab>Проекты</Tab>
            <Tab>Платежные системы</Tab>
          </TabList>

          <TabPanel>
            <TransactionsTable data={transactionsData} />
          </TabPanel>

          <TabPanel>
            <ProjectList transactionsData={transactionsData} />
          </TabPanel>

          <TabPanel>
            <PaymentMethodList paymentData={paymentData} />
            <PaymentMethodChart paymentData={paymentData} />
          </TabPanel>
        </Tabs>
      </div>
    )
  }
}

export default App