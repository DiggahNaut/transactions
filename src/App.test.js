import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
});

it("check App snapshot", () => {
    const AppComponent = renderer.create(<App />).toJSON();
    expect(AppComponent).toMatchSnapshot();
});