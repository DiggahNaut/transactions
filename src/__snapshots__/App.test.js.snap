// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check App snapshot 1`] = `
<div
  className="container"
>
  <div
    className="react-tabs"
    data-tabs={true}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <ul
      className="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-7"
        aria-disabled="false"
        aria-selected="true"
        className="react-tabs__tab react-tabs__tab--selected"
        id="react-tabs-6"
        role="tab"
        tabIndex="0"
      >
        Транзакции
      </li>
      <li
        aria-controls="react-tabs-9"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-8"
        role="tab"
        tabIndex={null}
      >
        Проекты
      </li>
      <li
        aria-controls="react-tabs-11"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-10"
        role="tab"
        tabIndex={null}
      >
        Платежные системы
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-6"
      className="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-7"
      role="tabpanel"
    >
      <table
        className="table table-striped table-hover"
      >
        <thead>
          <tr>
            <th
              className="table-cell"
              colSpan={4}
            >
              Список транзакций
            </th>
          </tr>
          <tr>
            <th
              className="table-cell"
              colSpan={1}
            >
              Проект
            </th>
            <th
              className="table-cell"
              colSpan={1}
            >
              Способ оплаты
            </th>
            <th
              className="table-cell"
              colSpan={1}
            >
              Пользователь
            </th>
            <th
              className="table-cell"
              colSpan={1}
            >
              Статус транзакции
            </th>
          </tr>
          <tr>
            <th
              colSpan={4}
            >
              <span>
                Поиск:
                 
                <input
                  className="table-filter"
                  onChange={[Function]}
                  placeholder="по всем записям"
                  value=""
                />
              </span>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Red Sauce 3
            </td>
            <td>
              Credit/Debit Cards
            </td>
            <td>
              Jack Black
            </td>
            <td>
              created
            </td>
          </tr>
          <tr>
            <td>
              Red Sauce 3
            </td>
            <td>
              QIWI
            </td>
            <td>
              e.polyanina
            </td>
            <td>
              canceled
            </td>
          </tr>
          <tr>
            <td>
              Red Sauce 3
            </td>
            <td>
              Credit/Debit Cards
            </td>
            <td>
              Jack Black
            </td>
            <td>
              created
            </td>
          </tr>
          <tr>
            <td>
              Red Sauce 3
            </td>
            <td>
              Credit/Debit Cards
            </td>
            <td>
              lester.mosley
            </td>
            <td>
              canceled
            </td>
          </tr>
          <tr>
            <td>
              Red Sauce 3
            </td>
            <td>
              Credit/Debit Cards
            </td>
            <td>
              e.polyanina
            </td>
            <td>
              canceled
            </td>
          </tr>
          <tr>
            <td>
              Red Sauce 3
            </td>
            <td>
              Credit/Debit Cards
            </td>
            <td>
              lester.mosley
            </td>
            <td>
              canceled
            </td>
          </tr>
          <tr>
            <td>
              Red Sauce 3
            </td>
            <td>
              Credit/Debit Cards
            </td>
            <td>
              lester.mosley
            </td>
            <td>
              canceled
            </td>
          </tr>
          <tr>
            <td>
              Red Sauce 3
            </td>
            <td>
              Credit/Debit Cards
            </td>
            <td>
              lester.mosley
            </td>
            <td>
              canceled
            </td>
          </tr>
          <tr>
            <td>
              God of Peace
            </td>
            <td>
              Credit/Debit Cards
            </td>
            <td>
              john.uniqe
            </td>
            <td>
              canceled
            </td>
          </tr>
          <tr>
            <td>
              Super Farm
            </td>
            <td>
              MobileGo
            </td>
            <td>
              karina.newman
            </td>
            <td>
              created
            </td>
          </tr>
        </tbody>
      </table>
      <div
        className="pagination"
      >
        <div>
          <button
            className="btn btn-secondary"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            &lt;&lt;
          </button>
           
          <button
            className="btn btn-secondary"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            &lt;
          </button>
           
          <button
            className="btn btn-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            &gt;
          </button>
           
          <button
            className="btn btn-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            &gt;&gt;
          </button>
        </div>
        <div>
          Страница
           
          1
           - 
          11
           
          из
           
          101
           
          записей
        </div>
        <div>
          <select
            className="form-control"
            onChange={[Function]}
            value={10}
          >
            <option
              value={5}
            >
              Показать 
              5
            </option>
            <option
              value={10}
            >
              Показать 
              10
            </option>
            <option
              value={15}
            >
              Показать 
              15
            </option>
            <option
              value={20}
            >
              Показать 
              20
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="react-tabs-8"
      className="react-tabs__tab-panel"
      id="react-tabs-9"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-10"
      className="react-tabs__tab-panel"
      id="react-tabs-11"
      role="tabpanel"
    />
  </div>
</div>
`;
